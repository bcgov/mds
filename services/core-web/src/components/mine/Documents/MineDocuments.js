/* eslint-disable */
import React, { Component } from "react";
import { bindActionCreators } from "redux";
import { connect } from "react-redux";
import { Divider } from "antd";
import AmazonS3Provider from "@/components/syncfusion/AmazonS3Provider";

/**
 * @class  MineDocuments - All documents pulled from MMS backdoor file system.
 */

const propTypes = {};

export class MineDocuments extends Component {
  state = {
    expandedKeys: [],
    searchValue: "",
    autoExpandParent: true,
  };

  onSelect = (keys, event) => {
    console.log("Trigger Select", keys, event);
  };

  onExpand = (expandedKeys) => {
    this.setState({
      expandedKeys,
      autoExpandParent: false,
    });
  };

  render() {
    return (
      <div className="tab__content">
        <div>
          <h2>Archived MMS Files</h2>
          <p>Archived MMS files are available on this page. These documents are read-only.</p>
          <Divider />
          <AmazonS3Provider />
        </div>
      </div>
    );
  }
}

const mapStateToProps = (state) => ({});

const mapDispatchToProps = (dispatch) => bindActionCreators({}, dispatch);

MineDocuments.propTypes = propTypes;

export default connect(mapStateToProps, mapDispatchToProps)(MineDocuments);
