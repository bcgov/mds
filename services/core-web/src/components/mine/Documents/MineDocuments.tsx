import React from "react";
import { connect } from "react-redux";
import { Divider } from "antd";
import { getMines, getMineGuid } from "@mds/common/redux/selectors/mineSelectors";
import AmazonS3Provider from "@mds/common/components/syncfusion/AmazonS3Provider";
import { RootState } from "@mds/common/redux/rootState";

interface MineDocumentsProps {
  mines: Record<string, { mine_no: string }>;
  mineGuid: string;
}

/**
 * @class MineDocuments - View the mine's archived MMS files.
 */
export const MineDocuments: React.FC<MineDocumentsProps> = (props: MineDocumentsProps) => {
  const mine = props.mines[props.mineGuid];
  return (
    <div className="tab__content">
      <div>
        <h2>Archived MMS Files</h2>
        <p>Archived MMS files are available on this page. These documents are read-only.</p>
        <Divider />
        <AmazonS3Provider mineNumber={mine.mine_no} />
      </div>
    </div>
  );
};

const mapStateToProps = (state: RootState) => ({
  mines: getMines(state),
  mineGuid: getMineGuid(state),
});

export default connect(mapStateToProps)(MineDocuments);
