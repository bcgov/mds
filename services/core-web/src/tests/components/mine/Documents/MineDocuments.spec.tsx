import React from "react";
import { shallow } from "enzyme";
import { MineDocuments } from "@/components/mine/Documents/MineDocuments";
import * as MOCK from "@/tests/mocks/dataMocks";

const setupProps = () => ({
  mines: MOCK.MINES.mines,
  mineGuid: "18133c75-49ad-4101-85f3-a43e35ae989a",
});

describe("MineDocuments", () => {
  it("renders properly", () => {
    const component = shallow(<MineDocuments {...setupProps()} />);
    expect(component).toMatchSnapshot();
  });
});
