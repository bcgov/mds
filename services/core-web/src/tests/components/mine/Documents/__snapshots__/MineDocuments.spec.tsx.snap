// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MineDocuments renders properly 1`] = `
<div
  className="tab__content"
>
  <div>
    <h2>
      Archived MMS Files
    </h2>
    <p>
      Archived MMS files are available on this page. These documents are read-only.
    </p>
    <Divider />
    <Connect(AmazonS3Provider)
      mineNumber="BLAH9091"
    />
  </div>
</div>
`;
